src/gui/owncloudsetupwizard.cpp

```diff
@@ -167,12 +167,15 @@ void OwncloudSetupWizard::slotFoundServer(const QUrl &url, const QJsonObject &in
         if (url.scheme() == QLatin1String("https") && oldUrl.host() == url.host()) {
             _ocWizard->account()->setUrl(url);
         } else {
-            auto accountState = new AccountState(_ocWizard->account());
-            connect(accountState, &AccountState::urlUpdated, this, [accountState, this] {
-                accountState->deleteLater();
-                slotDetermineAuthType();
-            });
-            accountState->updateUrlDialog(url);
+            auto accountState = AccountState::fromNewAccount(_ocWizard->account()).take(); // we manage the lifetime for this temporary AccountState ourselves (see below)
+            connect(accountState, &AccountState::urlUpdated, this, &OwncloudSetupWizard::slotDetermineAuthType);
```

Comment:
the connect could be moved down inside the if

Reply:
done